### PR TITLE
Add support for building literal set objects

### PR DESCRIPTION
**Pull request by jlowin**
Adds support for literal sets to our GQL parser

### CODE REVIEW

File: tests/utilities/test_graphql.py [hunk 1]
```diff
@@ -390,6 +391,13 @@ def test_enum_value_in_arguments():
     assert query == "where: { color: red }"
 
 
+def test_set_value_in_arguments():
+    query = parse_graphql_arguments(
+        {"where": {"color": LiteralSetValue(["red", "blue"])}}
+    )
+    assert query == 'where: { color: "{red, blue}" }'
```
Comment (cicdw) [inline tests/utilities/test_graphql.py]: Can you add a test with a single value?  Always an edge case

File: src/prefect/utilities/graphql.py [hunk 2]
```diff
@@ -45,6 +45,19 @@ def __str__(self) -> str:
         return self.value
 
 
+def LiteralSetValue(value: list) -> str:
+    """
+    When parsing GraphQL arguments, strings can be wrapped in this class to be rendered
```
Comment (cicdw) [inline src/prefect/utilities/graphql.py]: Not actually a class; maybe:
> When parsing GraphQL arguments, `LiteralSetValue` renders strings as literal set values...

Not really a big deal since this isn't truly a public API

Comment (jlowin) [reply src/prefect/utilities/graphql.py]: Ah thanks, I copied this from `EnumValue` because originally i implemented this as a class but oddly enough our parser has an easier time if it's just a prebuilt string.
